src/devices/machine/pit8253.cpp

```diff
@@ -1137,3 +1172,49 @@ void pit_counter_device::set_clock_signal(int state)
 	}
 	m_clock_signal = state;
 }
+
+uint8_t ps2_pit_device::read(offs_t offset)
+{
+	LOG2("read(): offset %d\n", offset);
+
+	if (offset == 3)
+	{
+		/* Reading mode control register is illegal according to docs */
+		/* Experimentally determined: reading it returns 0 */
+		return 0;
+	}
+	else if(offset == 4) // register 4 is timer channel 3.
+	{
+		return m_ch3_counter->read();
+	}
+	else if(offset <= 2)
+		return m_counter[offset]->read();
+	else
+		return 0xFF;
+}
+
+void ps2_pit_device::write(offs_t offset, uint8_t data)
+{
+	LOG2("write(): offset=%d data=0x%02x\n", offset, data);
+
+	if (offset == 3)
+	{
+		/* Write to mode control register */
+		int timer = (data >> 6) & 3;
+		if (timer == 3)
+			readback_command(data);
+		else
+			m_counter[timer]->control_w(data);
+	}
+	else if(offset == 7)
+	{
+		/* Write to mode control register */
+		m_ch3_counter->control_w(data);
+	}
+	else if(offset == 4)
+	{
+		m_ch3_counter->count_w(data);
+	}
+	else
+		m_counter[offset]->count_w(data);
+}
```

Comment:
Same here with `if (...)` vs `if(...)`.  Also, the file is missing its terminating newline.